synapse/lib/coro.py

```diff
@@ -248,20 +248,23 @@ def execspawn():
         proc.terminate()
         raise
 
-# shared process pool
 forkpool = None
 if multiprocessing.current_process().name == 'MainProcess':
-    # only create the forkpool in the MainProcess...
+    # only create the forkpools in the MainProcess...
     try:
         mpctx = multiprocessing.get_context('forkserver')
-        max_workers = int(os.getenv('SYN_FORKED_WORKERS', 1))
+        max_workers = int(os.getenv('SYN_FORKED_WORKERS', 0)) or min(4, os.cpu_count() or 4)
         forkpool = concurrent.futures.ProcessPoolExecutor(mp_context=mpctx, max_workers=max_workers)
+        logger.debug(f'Shared forkserver pool max_workers={forkpool._max_workers}')
```

Comment:
Is the intent that we log the # of workers upon service startup?

This will be logged during import time; which is prior to any logger configuration occurring.
If we set a default `max_workers = None` value than the workers is safe to refer to from other modules and log the # of as needed.

Reply:
We already had the logger.warning there, but yeah, it hasn't been configured yet.  I removed this debug logging, and maybe we should consider making the warning a python warning.